app/models/concerns/time_window_validator.rb

```diff
@@ -3,11 +3,13 @@ class TimeWindowValidator < ActiveModel::EachValidator
   MAX_DAYS_BACK = 29
 
   def validate_each(record, attribute, value)
-    record.errors.add attribute, (options[:message] || "exceeds the maximum of #{MAX_DAYS_BACK} days") unless recent_enough?(value)
```

Comment:
This would be incorrect. You would want to only return if recent_enough?(value) otherwise want to run record.errors.add().  So correct way is 
```
def validate_each(record, attribute, value)
  return if recent_enough?(value)

  record.errors.add(attribute, (options[:message] ||
                                "exceeds the maximum of #{MAX_DAYS_BACK} days"))
end
```